File: front/medicine/src/app/shared/models/viewModels/TokenData.ts
import { Language } from "../../enums/Language";
import { Role } from "../../enums/Role";
import { Sex } from "../../enums/Sex";
import { TimeZone } from "../../enums/Timezone";


export class TokenData {

  Email!: string;
  Id!: number;
  Language: Language = Language.en;
  Name!: string;

  TimeZone: TimeZone = TimeZone["(UTC) Coordinated Universal Time"];
  Role: Role = Role.user;
  Sex: Sex = Sex.male;
  Birthday!: Date;
  PhoneNumber!: string;
}

File: src/Web/Medicine.Web.UseCases/Common/TokenData.cs
using Medicine.Entities.Enums;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using TimeZone = Medicine.Entities.Enums.TimeZone;

namespace Medicine.Web.UseCases.Common
{


    public class TokenData
    {
        public string Email { get; set; }
        public int Id { get; set; }
        public Language Language { get; set; } = Language.en;
        public string Name { get; set; }
        public TimeZone TimeZone { get; set; } = TimeZone.CoordinatedUniversalTime;
        public SystemRole Role { get; private set; } = SystemRole.User;
        public Sex Sex { get; set; } = Sex.Male;
        public DateOnly? Birthday { get; set; }
        public string PhoneNumber { get; set; }

        public void SetRole (string role)
        {
            var asdfas = role;
        }
    }
}
